src/rust-2024/rpit-lifetime-capture.md

```diff
@@ -1,12 +1,313 @@
+- If a `use<..>` bound is elided (i.e. not present) then the compiler automatically inserts one according to edition-specific rules.
+- In Rust 2024, *all* in-scope generic parameters, including lifetime parameters, are included in these automatically inserted `use<..>` bounds.
+- Uses of the `Captures` trick (`Captures<..>` bounds) and of the outlives trick (e.g. `'_` bounds) can be replaced by `use<..>` bounds (in all editions) or removed entirely (in Rust 2024).
+
 ## Details
 
+### Capturing and `use<..>` syntax
+
+*Capturing* a generic parameter in an RPIT (return-position impl Trait) opaque type allows for that parameter to be used in the corresponding hidden type.  In Rust 1.82, we added `use<..>` bounds that allow specifying explicitly which generic parameters to capture.  E.g.:
+
+```rust
+# #![feature(precise_capturing)]
+fn capture<'a, T>(x: &'a (), y: T) -> impl Sized + use<'a, T> {
+    //                                ~~~~~~~~~~~~~~~~~~~~~~~
+    //                             This is the RPIT opaque type.
+    //
+    //                                It captures `'a` and `T`.
+    (x, y)
+  //~~~~~~
+  // The hidden type is: `(&'a (), T)`.
+  //
+  // This type can use `'a` and `T` because they were captured.
+}
+```
+
+Currently, `use<..>` bounds must list all generic type and const parameters in scope, but they may exclude in-scope lifetime parameters.  E.g.:
+
+```rust
+# #![feature(precise_capturing)]
+fn capture<'a, T>(_: &'a (), _: T) -> impl Sized + use<T> {
+    //                                ~~~~~~~~~~~~~~~~~~~
+    //                 The opaque type captures only `T` and not `'a`.
+    ()
+}
+```
+
+You might want to capture more lifetimes in order to return a hidden type that uses those lifetimes.  Conversely, you might want to capture fewer lifetimes in order to allow the opaque type to be used in more places.  E.g., this is an error because the lifetime is captured despite the fact that the hidden type does not use the lifetime:
+
+```rust,compile_fail
+# #![feature(precise_capturing)]
+fn capture<'a>(_: &'a ()) -> impl Sized + use<'a> {}
+
+fn test<'a>(x: &'a ()) -> impl Sized + 'static {
+    capture(x)
+    //~^ ERROR lifetime may not live long enough
+}
+```
+
+Conversely, this is OK since the opaque type does not capture the lifetime:
+
+```rust
+# #![feature(precise_capturing)]
+fn capture<'a>(_: &'a ()) -> impl Sized + use<> {}
+
+fn test<'a>(x: &'a ()) -> impl Sized + 'static {
+    capture(x) //~ OK
+}
+```
+
+The bounds list of an RPIT opaque type may contain at most one `use<..>` bound.  If this bound is present, then the listed generic parameters are the only ones captured.
+
+### Edition-specific rules for elided `use<..>` bounds
+
+If the `use<..>` bound is *elided* (i.e., not present), then the compiler automatically inserts one according to edition-specific rules.
+
+In all editions, all in-scope type and const generic parameters are added to this automatically-inserted `use<..>` bound.  E.g.:
```

Comment:
```suggestion
In all editions, all in-scope type and const generic parameters are captured.  E.g.:
```

Reply:
e.g.

Reply:
All descriptions of automatically adding bounds have been removed.